redaxo/src/addons/structure/pages/index.php

```diff
@@ -167,7 +167,12 @@
         $i_category_id = $KAT->getValue('id');
 
         $kat_link = $structureContext->getContext()->getUrl(['category_id' => $i_category_id]);
-        $kat_icon_td = '<td class="rex-table-icon"><a href="' . $kat_link . '" title="' . rex_escape($KAT->getValue('catname')) . '"><i class="rex-icon rex-icon-category"></i></a></td>';
+
+        $kat_object = rex_category::get($KAT->getValue('id'));
+        $kat_has_child_elements = (count($kat_object->getChildren()) > 0 || count($kat_object->getArticles()) > 1); // contains child categories or articles other than the start article
```

Comment:
```suggestion
        $kat_object = rex_category::get($KAT->getValue('id'));
        assert($kat_object instanceof rex_category);
        $kat_has_child_elements = (count($kat_object->getChildren()) > 0 || count($kat_object->getArticles()) > 1); // contains child categories or articles other than the start article
```

Das sollte das psalm-Problem lösen. Wir wissen, dass es die Kategorie gibt, da `$KAT` das zugehörige sql ist.
Für psalm machen wir so deutlich, dass das Kategorie-Objekt hier immer existiert, bzw. existieren muss.

Noch ein Hinweis: Ich versuche bei Variablen zur einheitlichen camelCase-Schreibweise zu kommen. Aktuell haben wir ein Mischmasch, aber es verschwinden mit der Zeit immer mehr die snake_case-Variablen.
Du musst wegen mir jetzt hier nicht die neuen Variablen ändern, aber so als Hinweis für zukünftige Änderungen, dass ich bei neuen Variablen die camelCase-Schreibweise bevorzuge.

Reply:
Danke für die Erklärung — Und okay, beim nächsten Mal: camelCase forTheWin!

Reply:
Args, scheint noch nicht zu passen. Ich werde es lokal austesten und das psalm problem lösen.